plugins/moz_databricks.py

```diff
@@ -0,0 +1,171 @@
+
+        See: https://docs.databricks.com/api/latest/jobs.html#runs-submit
+
+        :param job_name: Name of the job
+        :param env: Parameters via mozetl and tbv envvar wrappers
+        :param instance_count: The number of instances to use in production
+        :param dev_instance_count: The number of instances to use in development
+        :param max_instance_count: Max number of instances during autoscaling
+        :param dev_max_instance_count: Max number of instances during
+            autoscaling in dev
+        :param enable_autoscale: Enable autoscaling for the job
+        :param disable_on_dev: Turn the job into a no-op if run in development
+        :param release_label: Databricks Runtime versions,
+            run `databricks clusters spark-versions` for possible values.
+        :param iam_role: An Amazon Resource Name (ARN) specifying an iam role
+        :param owner: The e-mail address of the user owning the job.
+        :param uri: argument from EMRSparkOperator for compatibility
+        :param output_visibility: argument from EMRSparkOperator for compatibility
+
+        :param kwargs: Keyword arguments to pass to DatabricksSubmitRunOperator
+        """
+
+        if enable_autoscale:
+            if not max_instance_count:
+                raise ValueError("`max_instance_count` should be set when "
+                                 "`enable_autoscale` is enabled.")
+            if (max_instance_count < instance_count or
+                    dev_max_instance_count < dev_instance_count):
+                raise ValueError("The max instance count should be greater "
+                                 "than the instance count.")
+
+        is_dev = self.deploy_environment == 'dev'
+        self.disable_on_dev = disable_on_dev
+        self.job_name = job_name
+
+        jar_task = None
+        python_task = None
+        libraries = []
+
+        # Create the cluster configuration
+        new_cluster = {
+            "spark_version": release_label,
+            "node_type_id": self.instance_type,
+            "aws_attributes": {
+                "availability": "ON_DEMAND",
+                "instance_profile_arn": iam_role or self.default_iam
+
+            },
+            "spark_env_vars": env,
+            "custom_tags": {
+                "Owner": owner,
+                "Application": "databricks",
+                "Source": "Airflow",
+                "Job": job_name,
+            }
+        }
+
+        min_workers = dev_instance_count if is_dev else instance_count
+        max_workers = dev_max_instance_count if is_dev else max_instance_count
+        if enable_autoscale:
+            new_cluster["autoscale"] = {
+                "min_workers": min_workers,
+                "max_workers": max_workers,
+            }
+        else:
+            new_cluster["num_workers"] = min_workers
+
+        # Parse the environment variables to bootstrap the tbv/mozetl workflow
+        if env.get("TBV_CLASS"):
```

Comment:
Suggestion: instead of this if clause, use an abstract base class with the shared stuff and create two child classes, one each for jobs that use tbv_envvar and mozetl to set the bits in each clause

Reply:
I'm going to hold off on this one. I was actually thinking it might be interesting to think about a base class for Mozilla-specific information we would want to have in our jobs. The EMRSparkOperator should probably be moved into the plugins directory, and the common templating variables and constructor values could be moved into a common class.